### PR TITLE
docs: add `Range` example from Effekt

### PR DESCRIPTION
**Pull request by jaschdoc**
I don't know if this is too simple, but maybe it serves as a good introduction.

Related to https://github.com/flix/flix/issues/7093
Related to https://github.com/flix/flix/issues/7218

### CODE REVIEW

**Comment by jaschdoc** [comment]
This one is from benchmarks https://github.com/effect-handlers/effect-handlers-bench/blob/main/benchmarks/effekt/iterator/main.effekt

File: examples/effects/Range.flix [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+/// sums that range.
+///
+/// There are three things to pay attention to:
+/// - How to use effects to signal the caller
+/// - Handling the effect at the call site
+/// - Using the continuation to transfer control back to the callee
+///
+/// Inspired by the Iterator benchmark from Effekt Lang
+///
+pub def main(): Unit \ IO =
```
Comment (JonathanStarup) [inline examples/effects/Range.flix:15]: imo we should have `{ }` around function bodies longer than one line

File: examples/effects/Range.flix [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    pub def sumTo(n: Int32): Int32 = region rc {
+        let s = Ref.new(rc, 0);
+        try {
+            // Call an effectful function.
+            // Note that `sumTo` does not
+            // inherit the `Emission` effect type
+            // that `range` has, because we
+            // handle the effect below.
+            range(0, n);
+            Ref.get(s)
```
Comment (JonathanStarup) [inline examples/effects/Range.flix:52]: Do we prefer `Ref.{get, put}` instead of `!` and `:=`?

Comment (magnus-madsen) [reply examples/effects/Range.flix:52]: We prefer get and put because we are dropping those symbols later.

**Comment by JonathanStarup** [comment]
Its a bit of a weird use of effects but I think its actually a nice example to look at to learn about effect semantics and how it interacts with mutable state

**Comment by magnus-madsen** [comment]
I was going to suggest we take another look at these examples once Jonathans PR lands.

**Comment by magnus-madsen** [comment]
@jaschdoc Test failures

**Comment by jaschdoc** [comment]
Yeah just wanted to run tests to check what broke :)

**Comment by magnus-madsen** [comment]
> Yeah just wanted to run tests to check what broke :)

Probably removal of `do` everywhere is at least one thing.

**Review by magnus-madsen — APPROVED** [review]

**Pull request merged**